cmds/core/sluinit/uinit.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"github.com/u-root/u-root/pkg/cmdline"
+	slaunch "github.com/u-root/u-root/pkg/securelaunch"
+	"github.com/u-root/u-root/pkg/securelaunch/policy"
+	"github.com/u-root/u-root/pkg/securelaunch/tpm"
+	"log"
+	"os"
+)
+
+func main() {
+
+	slaunch.Debug("********Step 1: init completed. starting main ********")
+	tpmDev, err := tpm.GetHandle()
+	if err != nil {
+		log.Printf("tpm.getHandle failed. err=%v", err)
+		os.Exit(1)
+	}
+	defer tpmDev.Close()
+
+	slaunch.Debug("********Step 2: locate SL Policy ********")
+	rawBytes, err := policy.Locate()
+	if err != nil {
+		log.Printf("locate SL Policy failed: err=%v\n", err)
+		os.Exit(1)
+	}
+	slaunch.Debug("policy file located\n")
+
+	slaunch.Debug("********Step 3: parse SL Policy ********")
+	//TODO: The policy file must be measured and extended into PCR21 (PCR15
+	// until DRTM launch is working and able to set locality
+	p, err := policy.Parse(rawBytes)
+	if err != nil {
+		log.Printf("parse Policy failed")
+		os.Exit(1)
+	}
+	if p == nil {
+		log.Printf("SL Policy parsed into a null set")
+		os.Exit(1)
+	}
+	slaunch.Debug("policy file parsed\n")
+
+	slaunch.Debug("********Step 4: Collecting Evidence ********")
+	slaunch.Debug("policy file parsed=%v\n", p)
+
+	for _, c := range p.Collectors {
+		slaunch.Debug("Input Collector: %v\n", c)
+		c.Collect(tpmDev)
+	}
+	slaunch.Debug("Collectors completed\n")
+
+	slaunch.Debug("********Step 5: Write eventlog to /boot partition*********")
+	if e := p.EventLog.Persist(); e != nil {
+		log.Printf("EventLog.Persist() failed err=%v", e)
+		os.Exit(1)
+	}
+
+	slaunch.Debug("********Step 6: Launcher called ********")
+	err = p.Launcher.Boot(tpmDev)
+	log.Printf("Boot failed. err=%s", err)
+}
+
+func init() {
+	// check if uroot-debug=true in kernel cmdline.
+	val, ok := cmdline.Flag("uroot-debug")
+	if ok && val == "true" {
+		slaunch.Debug = log.Printf
+		slaunch.Debug("uroot-debug is set to true in kernel cmdline. Logging Enabled.")
+		slaunch.CmdExec("ls", []string{"/sys/class/net"})
```

Comment:
u-root/pkg/sh? sh.RunWithLogs("ls", "/sys/class/net")?

if sh is missing something you want, perhaps just add it there?

Reply:
well i was hoping to have one generic function in the entire securelaunch package ( 
slaunch.Debug) that only prints cmd stdout when the debug mode is on. also its cleaner to look
because i have them all over...

In this case, its already in the debug loop so we are okay, but in future...if i use sh.RunWithLogs...I would need to add additional variables which will be checked before calling sh.RunWithLogs..so that the log.Printfs inside sh.RunwithLogs only get printed in debug mode...i.e when that variable is true..

iif uroot-debug=true in kernel cmd line,
 /// set var = true

then check for that variable 
call sh.RunWithLogs... 

that would make the code filled with those checks....no ?

Reply:
I have decided to remove CmdExec for now. Two reasons:
1. Its only used to print network and block devices at init. We are already printing block devices via the storage package. and for these commits, network devices are not really needed. In future when we add support for scaning iscsi drives, we can add these debugs.
2. Lets find a way to make sh.RunWithLogs accept a debug var , so it only prints logs in debug mode...